src/main/java/jenkins/plugins/git/GitSCMFileSystem.java

```diff
@@ -252,19 +254,23 @@ public SCMFileSystem build(@NonNull Item owner, @NonNull SCM scm, @CheckForNull
                     git.using(tool.getGitExe());
                 }
                 GitClient client = git.getClient();
-                client.addDefaultCredentials(CredentialsMatchers.firstOrNull(
-                        CredentialsProvider.lookupCredentials(
-                                StandardUsernameCredentials.class,
-                                owner,
-                                ACL.SYSTEM,
-                                URIRequirementBuilder.fromUri(remote).build()
-                        ),
-                        CredentialsMatchers.allOf(
-                                CredentialsMatchers.withId(config.getCredentialsId()),
-                                GitClient.CREDENTIALS_MATCHER
-                        )
-                        )
-                );
+
+                StandardUsernameCredentials defaultsCreds = null;
+                String credsId = config.getCredentialsId();
+                if (credsId != null) {
```

Comment:
I want a review from @stephenc before I'm willing to accept this line does not change the logic of the code.  As far as I can tell, the earlier code always added default credentials, even if the config.getCredentialsId() returned null.  The new code does not call CredentialsProvider.lookupCredentials(..., owner, ...) if config.getCredentialsId() returns null.

The other changes in the pull request look good to me, but this one needs more review from a credentials expert.